pkg/docker/docker.go

```diff
@@ -23,3 +27,49 @@ func IsLocalHost(dockerHost string) bool {
 		// https://github.com/moby/moby/blob/master/client/client_unix.go#L6
 		strings.HasPrefix(dockerHost, "unix:")
 }
+
+// ClientOpts returns an appropiate slice of client.Opt values for connecting to a Docker client.
+// It can support using SSH connections via the Docker CLI's connection helpers if the DOCKER_HOST
+// environment variable is an SSH url, otherwise it will return client.FromEnv for a standard
+// connection. This function returns an error if DOCKER_HOST is an invalid URL.
+func ClientOpts() ([]client.Opt, error) {
+	connHelperOpts, err := connectionHelperOpts()
+	if err != nil {
+		return nil, err
+	}
+	if connHelperOpts != nil {
+		return connHelperOpts, nil
```

Comment:
would it make sense to pre-pend client.FromEnv to the beginning of this list? 

from what i can tell here:
https://github.com/docker/cli/blob/2b0e74c20773c43c2c21b50d5c8ee745e24c2746/cli/context/docker/load.go#L125

you still want to pull in options for docker api version negotiation.

Reply:
Yes I think you are right. I had originally done this, but thought it didn't work because of a red herring. I will push an update.